### PR TITLE
Add Zookeeper Standalone JMX Configuration

### PR DESCRIPTION
**Pull request by simplesteph**

### CODE REVIEW

File: example_configs/zookeeper.yaml [hunk 1]
```diff
@@ -15,4 +16,10 @@ rules:
     labels:
       replicaId: "$2"
       memberType: "$3"
-
+  # standalone Zookeeper
+  - pattern: "org.apache.ZooKeeperService<name0=StandaloneServer_port(\\d+)><>(.+)"
+    name: "zookeeper_$2"
+  - pattern: "org.apache.ZooKeeperService<name0=StandaloneServer_port(\\d+), name1=InMemoryDataTree><>(.+)"
+    name: "zookeeper_$2"
+    labels:
```
Comment (brian-brazil) [inline example_configs/zookeeper.yaml]: I'm not sure we should have this label

File: example_configs/zookeeper.yaml [hunk 2]
```diff
@@ -15,4 +16,10 @@ rules:
     labels:
       replicaId: "$2"
       memberType: "$3"
-
+  # standalone Zookeeper
+  - pattern: "org.apache.ZooKeeperService<name0=StandaloneServer_port(\\d+)><>(.+)"
```
Comment (brian-brazil) [inline example_configs/zookeeper.yaml]: This doesn't seem to be exactly what we have above.

**Comment by simplesteph** [comment]
@brian-brazil the right commit wasn't pushed. Now it's fixed!

**Comment by simplesteph** [comment]
I don't think there's value in others. I used Jolokia to list them. Started a standalone server and connected one client to it:

```
{
  "name0=StandaloneServer_port2181,name1=Connections,name2=127.0.0.1,name3=0x16523bb90320000": {
    "op": {
      "resetCounters": {
        "args": [],
        "ret": "void",
        "desc": "resetCounters"
      },
      "terminateSession": {
        "args": [],
        "ret": "void",
        "desc": "terminateSession"
      },
      "terminateConnection": {
        "args": [],
        "ret": "void",
        "desc": "terminateConnection"
      }
    },
    "attr": {
      "PacketsReceived": {
        "rw": false,
        "type": "long",
        "desc": "PacketsReceived"
      },
      "PacketsSent": {
        "rw": false,
        "type": "long",
        "desc": "PacketsSent"
      },
      "LastResponseTime": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "LastResponseTime"
      },
      "EphemeralNodes": {
        "rw": false,
        "type": "[Ljava.lang.String;",
        "desc": "EphemeralNodes"
      },
      "OutstandingRequests": {
        "rw": false,
        "type": "long",
        "desc": "OutstandingRequests"
      },
      "LastZxid": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "LastZxid"
      },
      "LastCxid": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "LastCxid"
      },
      "SourceIP": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "SourceIP"
      },
      "SessionTimeout": {
        "rw": false,
        "type": "int",
        "desc": "SessionTimeout"
      },
      "MinLatency": {
        "rw": false,
        "type": "long",
        "desc": "MinLatency"
      },
      "AvgLatency": {
        "rw": false,
        "type": "long",
        "desc": "AvgLatency"
      },
      "LastOperation": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "LastOperation"
      },
      "StartedTime": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "StartedTime"
      },
      "LastLatency": {
        "rw": false,
        "type": "long",
        "desc": "LastLatency"
      },
      "SessionId": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "SessionId"
      },
      "MaxLatency": {
        "rw": false,
        "type": "long",
        "desc": "MaxLatency"
      }
    },
    "class": "org.apache.zookeeper.server.ConnectionBean",
    "desc": "Information on the management interface of the MBean"
  },
  "name0=StandaloneServer_port2181,name1=InMemoryDataTree": {
    "op": {
      "approximateDataSize": {
        "args": [],
        "ret": "long",
        "desc": "approximateDataSize"
      },
      "countEphemerals": {
        "args": [],
        "ret": "int",
        "desc": "countEphemerals"
      }
    },
    "attr": {
      "WatchCount": {
        "rw": false,
        "type": "int",
        "desc": "WatchCount"
      },
      "NodeCount": {
        "rw": false,
        "type": "int",
        "desc": "NodeCount"
      },
      "LastZxid": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "LastZxid"
      }
    },
    "class": "org.apache.zookeeper.server.DataTreeBean",
    "desc": "Information on the management interface of the MBean"
  },
  "name0=StandaloneServer_port2181": {
    "op": {
      "resetLatency": {
        "args": [],
        "ret": "void",
        "desc": "resetLatency"
      },
      "resetMaxLatency": {
        "args": [],
        "ret": "void",
        "desc": "resetMaxLatency"
      },
      "resetStatistics": {
        "args": [],
        "ret": "void",
        "desc": "resetStatistics"
      }
    },
    "attr": {
      "PacketsReceived": {
        "rw": false,
        "type": "long",
        "desc": "PacketsReceived"
      },
      "PacketsSent": {
        "rw": false,
        "type": "long",
        "desc": "PacketsSent"
      },
      "MaxSessionTimeout": {
        "rw": true,
        "type": "int",
        "desc": "MaxSessionTimeout"
      },
      "AvgRequestLatency": {
        "rw": false,
        "type": "long",
        "desc": "AvgRequestLatency"
      },
      "NumAliveConnections": {
        "rw": false,
        "type": "long",
        "desc": "NumAliveConnections"
      },
      "TickTime": {
        "rw": true,
        "type": "int",
        "desc": "TickTime"
      },
      "MaxRequestLatency": {
        "rw": false,
        "type": "long",
        "desc": "MaxRequestLatency"
      },
      "StartTime": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "StartTime"
      },
      "ClientPort": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "ClientPort"
      },
      "OutstandingRequests": {
        "rw": false,
        "type": "long",
        "desc": "OutstandingRequests"
      },
      "MaxClientCnxnsPerHost": {
        "rw": true,
        "type": "int",
        "desc": "MaxClientCnxnsPerHost"
      },
      "MinRequestLatency": {
        "rw": false,
        "type": "long",
        "desc": "MinRequestLatency"
      },
      "Version": {
        "rw": false,
        "type": "java.lang.String",
        "desc": "Version"
      },
      "MinSessionTimeout": {
        "rw": true,
        "type": "int",
        "desc": "MinSessionTimeout"
      }
    },
    "class": "org.apache.zookeeper.server.ZooKeeperServerBean",
    "desc": "Information on the management interface of the MBean"
  }
}
```

**Comment by simplesteph** [comment]
@brian-brazil  I understand, but does Zookeeper Replicated also export connection information? If so, that may be an insane amount of metrics being exported if your Zookeeper has many connections, with very little upside to the each connection information's in my opinion. 

Just looking for your opinion on this, and I'll adapt the code if you still feel strongly about including name2 and name3

**Comment by brian-brazil** [comment]
> does Zookeeper Replicated also export connection information?

No idea, it didn't when I last checked but that was years ago.

> If so, that may be an insane amount of metrics being exported if your Zookeeper has many connections, with very little upside to the each connection information's in my opinion.

If we're exposing such information, we should stop doing so.

> I'll adapt the code if you still feel strongly about including name2 and name3

It's more I want to make sure we're being consistent, and not missing relevant metrics.

**Comment by simplesteph** [comment]
@brian-brazil , if you have a look at the existing metrics, it says:
```
pattern: "org.apache.ZooKeeperService<name0=ReplicatedServer_id(\\d+), name1=replica.(\\d+)><>(\\w+)"
```
The metrics are already filtered for `replica.` and therefore don't account for connections. 
In standalone, there won't be `replica.` prefix, so it'll be very hard to achieve consistency. 

As mentioned, I only found `name2` with connection information no `name3`. None contain the `replica.` prefix (expectedly) so I think it's consistent

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**